src/App.tsx

```diff
@@ -1,27 +1,183 @@
-import { useState } from "react";
+import { useEffect, useState } from "react";
 import "./App.css";
 import TodoForm, { ITodo } from "./components/TodoForm";
 import Todo from "./components/Todo";
+import { Trash } from "lucide-react";
+import toast, { Toaster } from "react-hot-toast";
 
 function App() {
-  const [todos, setTodos] = useState<ITodo[]>([]);
+  const [todos, setTodos] = useState<ITodo[]>(
+    JSON.parse(localStorage.getItem("todos")!) || []
+  );
+  const [status, setStatus] = useState<string>("all");
+  const [filteredTodos, setFilteredTodos] = useState<ITodo[]>([]);
+  const [toggleAllCompleted, setToggleAllCompleted] = useState(true);
+
+  useEffect(() => {
+    filterHandler();
+    localStorage.setItem("todos", JSON.stringify(todos));
+  }, [todos, status]);
+
+  useEffect(() => {
+    if (localStorage.getItem("todos") === null) {
+      localStorage.setItem("todos", JSON.stringify([]));
+    } else {
+      const localTodos = localStorage.getItem("todos");
+      setTodos(JSON.parse(localTodos!));
+    }
+  }, []);
+
+  // filter todos
+  const filterHandler = () => {
+    switch (status) {
+      case "completed":
+        setFilteredTodos(todos.filter((todo) => todo.completed));
+        break;
+      case "uncompleted":
+        setFilteredTodos(todos.filter((todo) => !todo.completed));
+        break;
+      default:
+        setFilteredTodos(todos);
+        break;
+    }
+  };
 
   // adding todo to the list
   function addTodo(todo: ITodo) {
     setTodos([todo, ...todos]);
+    const notify = () => toast.success("Todo added successfully!");
+    notify();
   }
 
   // deleting todo from the list
   function deleteTodo(id: string) {
-    setTodos(todos.filter((todo) => todo.id !== id));
+    // prompt("Are you sure to delete this todo?");
+    if (window.confirm("Are you sure to delete this todo?")) {
+      setTodos(todos.filter((todo) => todo.id !== id));
+      const notify = () => toast.success("Todo deleted successfully!");
+      notify();
+    }
+    return;
+  }
+
+  // toggle todo
+  function toggleTodo(id: string) {
+    setTodos(
+      todos.map((todo) => {
+        if (todo.id === id) {
+          return { ...todo, completed: !todo.completed };
+        }
+        return todo;
+      })
+    );
+  }
+
+  // update todo
+  function updateTodo(id: string, text: string) {
```

Comment:
You can move this fn updateTodo to utils file and add all the servies ike
addTodo,updateTodo,.....etc

You can replace localStorage with apis so we need only one file to make this logic